numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -1028,7 +1060,22 @@ PyArray_IntpFromIndexSequence(PyObject *seq, npy_intp *vals, npy_intp maxvals)
 NPY_NO_EXPORT int
 PyArray_IntpFromSequence(PyObject *seq, npy_intp *vals, int maxvals)
 {
-    return PyArray_IntpFromIndexSequence(seq, vals, (npy_intp)maxvals);
+    if (PyLong_CheckExact(seq) || !PySequence_Check(seq)) {
+        vals[0] = intp_from_scalar(seq);
+        if (PyErr_Occurred() != NULL) {
+            return -1;
+        } else {
```

Comment:
```suggestion
        }
        else {
```
Also in some other places.  It is fine, but just not NumPy C-style.

Reply:
Fixed in 6180fb68b20a967163eadfa47a80c1b526a0afe2